alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/ApiNodeContentSpringClient.java

```diff
@@ -0,0 +1,76 @@
+package eu.xenit.alfresco.webscripts.client.spring;
+
+import eu.xenit.alfresco.webscripts.client.spi.ApiNodeContentClient;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import lombok.SneakyThrows;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpMethod;
+import org.springframework.util.StreamUtils;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.HttpClientErrorException.NotFound;
+import org.springframework.web.client.HttpServerErrorException.InternalServerError;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+public class ApiNodeContentSpringClient implements ApiNodeContentClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(ApiNodeContentSpringClient.class);
+
+    private final RestTemplate restClient;
+    private final String url;
+
+    public ApiNodeContentSpringClient(AlfrescoProperties alfrescoProperties) {
+        this(alfrescoProperties, RestTemplateHelper.buildRestTemplate(alfrescoProperties));
+    }
+
+    public ApiNodeContentSpringClient(AlfrescoProperties alfrescoProperties, RestTemplate restTemplate) {
+        this.url = UriComponentsBuilder.fromHttpUrl(alfrescoProperties.getUrl())
+                .path("/service/api/node/content")
+                .toUriString();
+
+        this.restClient = restTemplate;
+    }
+
+    @Override
+    public boolean hasContent(String nodeRef) {
```

Comment:
don't you need to check which content-property ?

Reply:
See: `org.alfresco.repo.web.scripts.content.ContentInfo`:
```
    @Override
    public void execute(WebScriptRequest req, WebScriptResponse res) throws IOException
    {
        // create empty map of args
        Map<String, String> args = new HashMap<String, String>(0, 1.0f);
        // create map of template vars
        Map<String, String> templateVars = req.getServiceMatch().getTemplateVars();
        // create object reference from url
        ObjectReference reference = createObjectReferenceFromUrl(args, templateVars);
        NodeRef nodeRef = reference.getNodeRef();
        if (nodeRef == null)
        {
            throw new WebScriptException(HttpServletResponse.SC_NOT_FOUND, "Unable to find " + reference.toString());
        }

        // render content
        QName propertyQName = ContentModel.PROP_CONTENT;

        // Stream the content
        streamContent(req, res, nodeRef, propertyQName, false, null, null);
    }
```
that's how it works :man_shrugging: